### PR TITLE
[WIP] Finalise 2.0

### LINKED ISSUE #747: Final cleanup before 2.0 final
Just look through the docs, etc. and make sure everything looks good.

### PR DESCRIPTION
**Pull request by thesamovar**
Closes #747.

### CODE REVIEW

File: docs_sphinx/introduction/release_notes.rst [hunk 1]
```diff
@@ -1,8 +1,90 @@
...
+* Multiple event types. In addition to the default ``spike`` event, you can
+  create arbitrary events, and have these trigger code blocks (like reset)
+  or synaptic events. See :doc:`/advanced/custom_events`.
+* New units system allows arrays to have units. This eliminates the need for
+  a lot of the special casing that was required in Brian 1. See
+  :doc:`/user/units`.
+* Indexing variable by condition, e.g. you might write ``G.v['x>0']`` to
+  return all values of variable ``v`` in `NeuronGroup` ``G`` where the
+  group's variable ``x>0``.
+* Correct numerical integraton of stochastic differential equations.
```
Comment (mstimberg) [inline docs_sphinx/introduction/release_notes.rst]: integraton --> integration

File: docs_sphinx/introduction/release_notes.rst [hunk 2]
```diff
@@ -1,8 +1,90 @@
...
+  See :doc:`/user/numerical_integration`.
+* "Magic" `run` system has been greatly simplified and is now much more
+  transparent. In addition, if there is any ambiguity about what the user
+  wants to run, an erorr will be raised rather than making a guess. This
+  makes it much safer. See :doc:`/user/running`.
```
Comment (mstimberg) [inline docs_sphinx/introduction/release_notes.rst]: I'd also mention the store/restore mechanism (to/from memory and disk) here

File: docs_sphinx/introduction/release_notes.rst [hunk 3]
```diff
@@ -1,8 +1,90 @@
...
+* New system for handling externally defined functions. They have to specify
+  which units they accept in their arguments, and what they return. In
+  addition, you can easily specify the implementation of user-defined
+  functions in different languages for code generation. See
+  :doc:`/advanced/functions`.
+* New generalised "flags" system for giving additional information when
+  defining models. See :ref:`flags`.
+* `TimedArray` now allows for 2D arrays with arbitrary indexing.
```
Comment (mstimberg) [inline docs_sphinx/introduction/release_notes.rst]: I think technically that already worked in Brian 1.

File: docs_sphinx/introduction/release_notes.rst [hunk 4]
```diff
@@ -1,8 +1,90 @@
...
+
+* Multicompartmental modelling.
+
+* Python 2 and 3 support.
+
+New features
+~~~~~~~~~~~~
+
+* Many improvements to `Synapses` which replaces the old ``Connection``
+  object in Brian 1. This includes: multiple synapses between a pair of neurons;
```
Comment (mstimberg) [inline docs_sphinx/introduction/release_notes.rst]: Brian 1 already had multiple synapses support.

File: docs_sphinx/introduction/release_notes.rst (hunk 3 above)
Comment (thesamovar) [inline docs_sphinx/introduction/release_notes.rst]: Not with the indexing I think!

File: docs_sphinx/introduction/release_notes.rst (hunk 4 above)
Comment (thesamovar) [inline docs_sphinx/introduction/release_notes.rst]: Ah, OK, my bad. Will remove.

**Comment by thesamovar** [comment]
OK, I'm done with this. I wonder if we should have a contributions ordered by number of commits for the whole of Brian 2 (i.e. the whole history of the repository)? I'll leave you to add that. We should maybe also credit Ben who contributed code to the ``brian2-binder`` repo but not to the main repo?

File: docs_sphinx/introduction/release_notes.rst (hunk 3 above)
Comment (mstimberg) [inline docs_sphinx/introduction/release_notes.rst]: Ah, you are right. Or rather I think you could do something like that but it worked rather by accident (and not with codegen options switched on I guess) -- at least it was not documented, so let's leave it in.

**Comment by mstimberg** [comment]
> We should maybe also credit Ben who contributed code to the brian2-binder repo but not to the main repo?

Yes, I'm happy to just count the commits to `brian2-binder` as `brian2` commits.

I'll add the commit statistics.

**Comment by mstimberg** [comment]
I added contributors to the release notes (and in a new AUTHORS file) and updated the readme and setup.py. For the full authors list I decided to go for "Authors by date of their first commit"), I think this is less misleading than the number of commits and we can more easily keep AUTHORS up-to-date (each new author will be added to the end). If you are happy with all this, feel free to go ahead and merge, everything looks good from my side otherwise.

**Pull request merged**